### DISCUSSION TITLE
Need to add `updatedAt` field to several tables that already contain data

### DISCUSSION THREAD

**Discussion by erasebegin** [prisma/prisma#16464] [Q&A] [answered]
When I try to migrate the schema after making this change, I receive the following error:

```
Step 0 Added the required column `updatedAt` to the `Person` table without a default value. There are 17 rows in this table, it is not possible to execute this step.
```

This recurs for every table that I am trying to modify by adding the `updatedAt` field (`updatedAt DateTime @updatedAt`).

The output suggests that I use `prisma migrate dev --create-only` which I guess will create a SQL file which I would then have to edit myself, but I don't know any SQL so I don't think that will go very well. 

I would prefer not to have to drop the rows of these tables, but it's not totally out of the question as it's only dummy data at the moment. Would just be a pain to recreate. To my mind it seems like it should be possible to add a new column to a table without losing the current data. Am I able to somehow provide a default value that would be added to all of the preexisting rows, as suggested by the error message? 

Any advice would be greatly appreciated 🙏

**Comment by janpio** [comment, marked as answer]
You can probably do it in multiple steps: Add the field as optional, run a script to fill in the default values you want to have, then add the `@updatedAt` attribute when all fields already have a value in that column.

**Reply by erasebegin** [reply]
Ahh that’s a really good suggestion, thank you! I didn’t even think about making it optional initially. That should do the trick! 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
